pkg/cmd/admin/node/listpods.go

```diff
@@ -61,8 +68,40 @@ func (l *ListPodsOptions) runListPods(node *kapi.Node, printer kubectl.ResourceP
 	if err != nil {
 		return err
 	}
+
 	fmt.Fprint(l.Options.ErrWriter, "\nListing matched pods on node: ", node.ObjectMeta.Name, "\n\n")
 	printer.PrintObj(pods, l.Options.Writer)
 
 	return err
 }
+
+// handleRESTOutput receives a list of nodes, and a REST output type, and combines *kapi.PodList
+// objects for every node, into a single list. This allows output containing multiple nodes to be
+// printed to a single writer, and be easily parsed as a single data format.
+func (l *ListPodsOptions) handleRESTOutput(nodes []*kapi.Node, printer kubectl.ResourcePrinter) []error {
+	var unifiedPodList *kapi.PodList
+
+	errList := []error{}
+	for _, node := range nodes {
+		labelSelector, err := labels.Parse(l.Options.PodSelector)
+		if err != nil {
+			errList = append(errList, err)
+		}
+		fieldSelector := fields.Set{GetPodHostFieldLabel(node.TypeMeta.APIVersion): node.ObjectMeta.Name}.AsSelector()
+
+		pods, err := l.Options.Kclient.Pods(kapi.NamespaceAll).List(kapi.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSelector})
+		if err != nil {
+			errList = append(errList, err)
+		}
+
+		if unifiedPodList == nil {
+			unifiedPodList = pods
```

Comment:
careful about using the first list as the aggregate... not sure the selfLink or resourceVersion will be right